tests/modules/geometry-calculation.test.js

```diff
@@ -0,0 +1,104 @@
+const Line = require('../../modules/ecs6-class/line');
+const Point = require('../../modules/ecs6-class/point');
+const { calculateDistance, calculateJunctionPoint, isPointOnLine } = require('../../modules/geometry-calculation')
+
+describe('geometry-calculation', () => {
+    const mockConstructorLine = jest.fn(constructor)
+    const mockConstructorPoint = jest.fn(constructor)
+    
```

Comment:
this is not how to mock a class, only a constructor.
you could have written only one mockFunction for the constructor,
it could have been used for all constructors: line, point and more

write real mocks for a class

Reply:
Is this a correct direction for mock, or does it not even come close?

const mockPoint = {
 x: 0,
 y: 0,
 moveVertical: sinon.stub().callsFake((value) => {
 console.log(value);
 return mockPoint.y + value;
 }),
 moveHorizontal: sinon.stub().callsFake((value) => {
 return mockPoint.x + value;
 })
 };

Reply:
don't use the sinon object to mock
use jest.mock('module', ()=>{})
there are a few ways
search build mock tests with jest, mock es6 class with jest, mock partial class es6 jest, __esmodule